clients/privacy-center/cypress/e2e/consent-banner.cy.ts

```diff
@@ -0,0 +1,181 @@
+        }
+    };
+
+    Cypress.Commands.add("visitConsentDemo", (consent?: ConsentConfig, bannerOptions?: ConsentBannerOptions) => {
+        cy.visit("/fides-js-components-demo.html", {
+            onBeforeLoad: (win) => {
+                // eslint-disable-next-line no-param-reassign
+                win.fidesConsentBannerOptions = {
+                    consent: {
+                        ...testBannerOptions.consent,
+                        ...consent,
+                    },
+                    bannerOptions: {
+                        ...testBannerOptions.bannerOptions,
+                        ...bannerOptions,
+                    }
+                }
+            },
+        });
+    })
+
+    describe("when disabled", () => {
+        beforeEach(() => {
+            cy.visitConsentDemo(undefined, {isDisabled: true });
+        });
+
+        it("does not render", () => {
+            cy.get("div#fides-consent-banner").should("not.exist");
+            cy.contains("button", "Accept Test").should("not.exist");
+        });
+    })
+
+    describe("when user has no saved consent cookie", () => {
+        beforeEach(() => {
+            cy.getCookie(CONSENT_COOKIE_NAME).should("not.exist");
+        });
+
+        describe("when banner is not disabled", () => {
+            beforeEach(() => {
+                cy.visitConsentDemo(undefined, {
+                    isDisabled: false,
+                    isGeolocationEnabled: false,
+                });
+            });
+
+            it("should render the expected HTML banner", () => {
+                // Consider me paranoid, but the various DOM APIs that consent-banner.ts uses
+                // (like createElement) are very easy to mess up. This test is (almost) a DOM
+                // snapshot test that asserts that the rendered banner HTML is similar to this:
+                //
+                // <div id='fides-consent-banner' class='fides-consent-banner'>
+                //   <div id='fides-consent-banner-description' class='fides-consent-banner-description'>
+                //     {labels.bannerDescription}
+                //   </div>
+                //   <div id='fides-consent-banner-buttons' class='fides-consent-banner-buttons'>
+                //     <button id='fides-consent-banner-tertiary-button' class='fides-consent-banner-button fides-consent-banner-primary-button'>
+                //       {labels.tertiaryButton}
+                //     </button>
+                //     <button id='fides-consent-banner-secondary-button' class='fides-consent-banner-button fides-consent-banner-secondary-button'>
+                //       {labels.secondaryButton}
+                //     </button>
+                //     <button id='fides-consent-banner-primary-button' class='fides-consent-banner-button fides-consent-banner-primary-button'>
+                //       {labels.primaryButton}
+                //     </button>
+                //   </div>
+                // </div>
```

Comment:
This comment feels unnecessary now - it's no longer as fragile (thank you!) so what this test should do is just help us double-check the render behaviour stays consistent as we continue to tinker with the internals.